src/main/java/org/openrewrite/java/migrate/util/OptionalNotEmptyToIsPresent.java

```diff
@@ -55,29 +55,29 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         TreeVisitor<?, ExecutionContext> check = Preconditions.and(
                 new UsesJavaVersion<>(11),
                 new UsesMethod<>(JAVA_UTIL_OPTIONAL_IS_EMPTY));
-        MethodMatcher optionalIsPresentMatcher = new MethodMatcher(JAVA_UTIL_OPTIONAL_IS_EMPTY);
         return Preconditions.check(check, new JavaVisitor<ExecutionContext>() {
             @Override
-            public Statement visitStatement(Statement s, ExecutionContext ctx) {
-                Statement statement = (Statement) super.visitStatement(s, ctx);
-                if (statement instanceof J.Unary) {
-                    J.Unary unary = (J.Unary) statement;
+            public J visitStatement(Statement s, ExecutionContext ctx) {
+                J superReturn = super.visitStatement(s, ctx);
+                if (superReturn instanceof J.Unary) {
+                    J.Unary unary = (J.Unary) superReturn;
                     if (unary.getOperator() == Type.Not) {
                         Expression expression = unary.getExpression();
                         if (expression instanceof J.MethodInvocation) {
                             J.MethodInvocation m = (J.MethodInvocation) expression;
+                            MethodMatcher optionalIsPresentMatcher = new MethodMatcher(JAVA_UTIL_OPTIONAL_IS_EMPTY);
                             if (optionalIsPresentMatcher.matches(m)) {
                                 return JavaTemplate.builder("#{any()}.isPresent()")
                                         .contextSensitive()
                                         .build()
                                         .apply(getCursor(),
-                                                statement.getCoordinates().replace(),
+                                                unary.getCoordinates().replace(),
                                                 m.getSelect());
                             }
                         }
                     }
                 }
-                return statement;
+                return superReturn;
```

Comment:
Reading the issue an alternative I'd considered was to just move the `super.visitStatement` here. I don't think it's likely that we encounter double nested negation with this recipe, and might simplify the logic above.

Reply:
AFAIK calling the super method in these visitors is to allow _any other_ potential optimization inside. So `!(somethingThatCanBeOptimizedToIsEmpty())` would be affected by that change since it would no longer be optimized in one iteration (but still by calling rewrite 2 times). However, I'm not that deep into the general visitor architecture of the OpenRewrite visitors, and I might have understood that wrong. (e.g. the recursion could be implemented fully outside any of the visitors by just having some manager checking whether the return value of the visitor has changed, and then just running all recipes again on that return node).
So if you want it changed that way, I can do it.

Reply:
I'm having a bit of trouble thinking of a scenario we'd miss when we only call the super method on line 80, and only in a very narrow case of  `!optional.isEmpty()` return from line 70. Any other cases would still fall through to call the super method then right? Note that I haven't opened my IDEA today, so perhaps I'm missing something.